source/_patterns/01-elements/input/input.hbs

```diff
@@ -1,25 +1,33 @@
-<!-- The aria-labelledby attribute on the input and related id on the label as well as aria-hidden="true" are being set for correct support within assistive tools like screenreaders //-->
-<!-- [html-validate-disable-next input-missing-label] -->
-<input type="{{ type }}"
-	class="elm-input {{ styleModifier }}"
-	placeholder="{{ placeholder }}"
-	name="{{ id }}"
-	id="{{ id }}"
-	{{#if disabled}} disabled{{/if }}
-	value="{{ value }}"
-	{{#if describedbyid }} aria-describedby="{{ describedbyid }}"{{/if }}
-	{{#if aria-invalid }} aria-invalid="{{ aria-invalid }}"{{/if }}
-	{{{ validation }}}
-	{{#if datalist }} list="datalist{{ id }}"{{/if }}
-	aria-labelledby="{{ id }}-label"
-	{{#if variant}} data-variant="{{variant}}"{{/if }}>
-<label class="elm-label"
-	for="{{ id }}"
-	aria-hidden="true"
-	id="{{ id }}-label">{{ label }}</label>
-{{#if description }}<p{{#if describedbyid }} id="{{ describedbyid }}"{{/if }} class="description">{{{ description }}}</p>{{/if }}
-{{#if datalist }}<datalist id="datalist{{ id }}">
-	{{#each datalistItems }}
-		<option{{#if value }} value="{{ value }}"{{/if }}>{{ text }}</option>
-	{{/each }}
-</datalist>{{/if }}
+<div class="elm-input"
+	{{#if variant}} data-variant="{{variant}}"{{/if }}
+	{{#if icon-before}} data-icon-before="{{icon-before}}"{{/if}}
+	{{#if icon-before}} data-icon-after="{{icon-after}}"{{/if}}>
+	<!-- The aria-labelledby attribute on the input and related id on the label as well as aria-hidden="true" are being set for correct support within assistive tools like screenreaders //-->
+	<!-- [html-validate-disable-next input-missing-label] -->
+	<input type="{{ type }}"
+		{{#if styleModifier }}class="{{ styleModifier }}"{{/if }}
+		placeholder="{{ placeholder }}"
+		name="{{ id }}"
+		id="{{ id }}"
+		{{#if disabled}} disabled{{/if }}
+		value="{{ value }}"
+		{{#if describedbyid }} aria-describedby="{{ describedbyid }}"{{/if }}
+		{{#if aria-invalid }} aria-invalid="{{ aria-invalid }}"{{/if }}
+		{{{ validation }}}
+		{{#if datalist }} list="datalist{{ id }}"{{/if }}
+		aria-labelledby="{{ id }}-label">
+	<label for="{{ id }}"
+		aria-hidden="true"
+		id="{{ id }}-label">{{ label }}</label>
+
+	{{#if description }}<p{{#if describedbyid }} id="{{ describedbyid }}"{{/if }} class="description">{{{ description }}}</p>{{/if }}
+
+	{{!-- if span elm-icon o.ä. mit data-icon --}}
+	{{#if datalist }}<datalist id="datalist-{{ id }}">
+		{{#each datalistItems }}
+			<option{{#if value }} value="{{ value }}"{{/if }}>{{ text }}</option>
+		{{/each }}
+	</datalist>
+	{{/if }}
+	<span class="elm-state-icon"></span>
```

Comment:
Couldn't we use the `icon()` SCSS mixin related to the `data-variant` value on the surrounding `div.elm-input` instead of adding an additional HTML tag ?

Reply:
As discussed we moved this discussion to https://github.com/db-ui/core/issues/223